tests/test_tokenization_common.py

```diff
@@ -200,19 +201,13 @@ def setUp(self) -> None:
         # Tokenizer.filter makes it possible to filter which Tokenizer to case based on all the
         # information available in Tokenizer (name, rust class, python class, vocab key name)
         if self.test_rust_tokenizer:
-            tokenizers_list = [
+            self.tokenizers_list = [
                 (
                     self.rust_tokenizer_class,
-                    pretrained_name,
+                    self.from_pretrained_id,
                     self.from_pretrained_kwargs if self.from_pretrained_kwargs is not None else {},
                 )
-                for pretrained_name in self.rust_tokenizer_class.pretrained_vocab_files_map[
-                    self.from_pretrained_vocab_key
-                ].keys()
-                if self.from_pretrained_filter is None
-                or (self.from_pretrained_filter is not None and self.from_pretrained_filter(pretrained_name))
             ]
-            self.tokenizers_list = tokenizers_list[:1]  # Let's just test the first pretrained vocab for speed
```

Comment:
@amyeroberts the gist of the changes is what is happening here. So far we're relying on this `self.rust_tokenizer_class.pretrained_vocab_files_map`, which returns the list of defined checkpoints from the vocab files maps.

We then only select the first one: `self.tokenizers_list = tokenizers_list[:1] `

With this change, we need to define, within testing, the repo ID that we'll use for integration tests. IMO it's much cleaner this way as we're not putting a list of random repo IDs within the user's runtime. This frees the way for the #29112 refactor to ship.

Reply:
Agreed - much cleaner!